### PR TITLE
Update ignoring.rst

### PR DESCRIPTION
**Pull request by blindstitch**
Converted to tabular format for readability. Ignore pattern characters and behaviors are separated to make the text easier to scan.

### CODE REVIEW

File: users/ignoring.rst [hunk 1]
```diff
@@ -34,62 +34,144 @@ Patterns
...
-.. note::
-
-   Prefixes can be specified in any order (e.g. "(?d)(?i)"), but cannot be in a
+Regular file names match themselves, i.e. the pattern foo matches the files foo,
+subdir/foo as well as any directory named foo. Spaces are treated as regular characters.
+
++-----------------+-----------------+-----------------+-----------------+
+| Character       | Where           | Affects         | Action          |
++=================+=================+=================+=================+
+| ``/``           | Beginning       | Folders         | Match in the    |
```
Comment (calmh) [inline users/ignoring.rst]:
```suggestion
| ``/``           | Beginning       | All         | Match in the    |
```

File: users/ignoring.rst [hunk 2]
```diff
@@ -34,62 +34,144 @@ Patterns
...
+|                 |                 |                 | root directory  |
+|                 |                 |                 | only. ``/foo``  |
+|                 |                 |                 | matches foo but |
+|                 |                 |                 | not             |
+|                 |                 |                 | ``subdir/foo``. |
++-----------------+-----------------+-----------------+-----------------+
+| ``!``           | Beginning       | Folders         | Ignore other    |
```
Comment (calmh) [inline users/ignoring.rst]:
```suggestion
| ``!``           | Beginning       | All         | Ignore other    |
```

**Comment by blindstitch** [comment]
> Did you use some tool for this? The resulting table looks like a royal pain in the ass to edit manually in the future...

It was a markdown table converted to RST format with pandoc. There might be a better table format in RST, I don't know. I modified the commit to make it easier to modify in a text editor.

File: users/ignoring.rst [hunk 3]
```diff
@@ -34,62 +34,63 @@ Patterns
...
-
--  A line beginning with ``//`` is a comment and has no effect.
-
--  Windows does not support escaping ``\[foo - bar\]``.
-
-.. note::
-
-   Prefixes can be specified in any order (e.g. "(?d)(?i)"), but cannot be in a
+Regular file names match themselves, i.e. the pattern foo matches the files foo,
+subdir/foo as well as any directory named foo. Spaces are treated as regular characters.
```
Comment (calmh) [inline users/ignoring.rst]:
```suggestion
Regular file names match themselves, i.e. the pattern foo matches the files ``foo``,
``subdir/foo`` as well as any directory named ``foo``. Spaces are treated as regular characters.
```

File: users/ignoring.rst [hunk 4]
```diff
@@ -34,62 +34,63 @@ Patterns
...
+
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| Character    | Where     | Affects  | Action                                                                            |
++==============+===========+==========+===================================================================================+
+| ``/``        | Beginning | Folders  | Match in the root directory only. ``/foo`` matches foo but not ``subdir/foo``.    |
```
Comment (calmh) [inline users/ignoring.rst]:
```suggestion
| ``/``        | Beginning | Folders  | Match in the root directory only. ``/foo`` matches ``foo`` but not ``subdir/foo``.    |
```

File: users/ignoring.rst [hunk 5]
```diff
@@ -34,62 +34,63 @@ Patterns
...
+|              |           |          |                                                                                   |
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``!``        | Beginning | All      | Ignore all other patterns and force sync of matching files. Matching directory    |
```
Comment (calmh) [inline users/ignoring.rst]: Original description is more correct, "all other" patterns aren't ignored

File: users/ignoring.rst [hunk 6]
```diff
@@ -34,62 +34,63 @@ Patterns
...
+|              |           |          | trees will be synced entirely regardless of other ignore patterns.                |
+|              |           |          |                                                                                   |
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``?d``       | Beginning | Files    | Files with names matching the pattern will be removed if they prevent directory   |
```
Comment (calmh) [inline users/ignoring.rst]: The ?d and similar must be parenthesized, examples are incorrect

File: users/ignoring.rst [hunk 7]
```diff
@@ -34,62 +34,63 @@ Patterns
...
+|              |           |          | deletion. Examples: ``?.DS_Store``, ``?.Recycle``                                 |
+|              |           |          |                                                                                   |
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``?i``       | Beginning | All      | Enable case-insensitiv e pattern matching. The (?i) prefix can be combined with   |
```
Comment (calmh) [inline users/ignoring.rst]: Parens, and `case insensitive`

File: users/ignoring.rst [hunk 8]
```diff
@@ -34,62 +34,63 @@ Patterns
...
+|              |           |          | other patterns, for example the pattern ``(?i)!picture* .png`` indicates that     |
+|              |           |          | Picture1.PNG should be synchronized. On Mac OS and Windows, patterns are always   |
+|              |           |          | case-insensitive.                                                                 |
+|              |           |          |                                                                                   |
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``//``       | Beginning | N/A      | Comments. Contents of the line will have no effect.                               |
+|              |           |          |                                                                                   |
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``#include`` | Beginning | N/A      | Load sync patterns from the referenced file. Including a file more than once,     |
```
Comment (calmh) [inline users/ignoring.rst]: Not "sync patterns"

File: users/ignoring.rst [hunk 9]
```diff
@@ -34,62 +34,63 @@ Patterns
...
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``**``       | Anywhere  | All      | Match every character including directory separators. ``te**st`` matches          |
+|              |           |          | ``test``, ``subdir/telere st`` and ``tele/sub/dir/ rest``.                        |
+|              |           |          |                                                                                   |
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``?``        | Anywhere  | All      | Match a single character that is not the directory separator.                     |
+|              |           |          |                                                                                   |
++--------------+-----------+----------+-----------------------------------------------------------------------------------+
+| ``[]``       | Anywhere  | All      | Acts like regex character range ``[a-z]`` matches all alphabet characters.        |
```
Comment (calmh) [inline users/ignoring.rst]: Sentence structure

**Comment by blindstitch** [comment]
> There are a lot of changes in the description, some of which are now incorrect. See specific notes, and please double check that all relevant meaning is carried over correctly from the original in this change. It sounded like just a format change, but it's much more than that...

In my zeal I rewrote them, dunno what I was thinking. I put back in the original descriptions.

File: users/ignoring.rst [hunk 10]
```diff
@@ -34,62 +34,77 @@ Patterns
...
+|             |           |          | ``/foo`` matches ``foo`` but not ``subdir/foo``.                              |
+|             |           |          |                                                                               |
++-------------+-----------+----------+-------------------------------------------------------------------------------+
+| ``!``       | Beginning | All      | A pattern beginning with a ``!`` prefix negates the pattern: matching files   |
+|             |           |          | are *included* (that is, *not* ignored). This can be used to override         |
+|             |           |          | more general patterns that follow.                                            | 
+|             |           |          | Include patterns (that begin with ``!``) cause Syncthing to traverse and      |
+|             |           |          | :ref:`watch <scanning>` the entire directory tree regardless of other         |
+|             |           |          | ignore patterns.                                                              |
```
Comment (imsodin) [inline users/ignoring.rst]: That could be misunderstood or at least confuse as to whether the other patterns apply still. I'd just put a reference to the note below here. And then in the note explicitly state that the ignore patterns still apply to the results of scanning/watching.

File: users/ignoring.rst [hunk 11]
```diff
@@ -34,62 +34,77 @@ Patterns
...
-.. note::
-
-   Prefixes can be specified in any order (e.g. "(?d)(?i)"), but cannot be in a
+Regular file names match themselves, i.e. the pattern foo matches the files ``foo``,
+``subdir/foo`` as well as any directory named ``foo``. Spaces are treated as regular characters.
+
++-------------+-----------+----------+-------------------------------------------------------------------------------+
+| Character   | Where     | Affects  | Action                                                                        |
++=============+===========+==========+===============================================================================+
+| ``/``       | Beginning | Folders  | A pattern beginning with ``/`` matches in the current directory only.         |
```
Comment (imsodin) [inline users/ignoring.rst]: Not sure what you mean by "Affects: Folders". And this is the only item that is not either `All` or `N/A`, so I also don't get the point of this column.

File: users/ignoring.rst [hunk 12]
```diff
@@ -34,62 +34,77 @@ Patterns
...
+|             |           |          |                                                                               |
++-------------+-----------+----------+-------------------------------------------------------------------------------+
+| ``?``       | Anywhere  | All      | Question mark matches a single character that is not the directory            |
+|             |           |          | separator. ``te??st`` matches ``tebest`` but not ``teb/st`` or                |
+|             |           |          | ``test``.                                                                     |
+|             |           |          |                                                                               |
++-------------+-----------+----------+-------------------------------------------------------------------------------+
+| ``[]``      | Anywhere  | All      | Characters enclosed in square brackets ``[]`` are interpreted as a character  |
+|             |           |          | range ``[a-z]``. Before using this syntax you should have a basic             |
+|             |           |          | understanding of regular expression character classes.                        |
```
Comment (imsodin) [inline users/ignoring.rst]: I know this is not your sentence, but I'd still remove it while at it: It insinuates globbing supports character classes (`\s`, `\W` or similar) while I am pretty sure it does not. And just generally I'd like to not mention regex here, as our syntax is explicitly not regex.

**Comment by st-review** [comment]
:robot: *beep boop*

I'm going to close this pull request as it has been idle for more than 90 days.

This is not a rejection, merely a removal from the list of active pull requests that is periodically reviewed by humans. The pull request can be reopened when there is new activity, and merged once any remaining issues are resolved.

**Pull request closed without merging**